src/types/ErrorRate/utils.ts

```diff
@@ -115,3 +121,62 @@ export const aggregate = (
   }
   return result;
 };
+
+export const getRateHealthConfig = (ns: string, name: string, kind: string, config?: string): ToleranceConfig[] => {
+  if (config && config !== '') {
+    const tolerances: ToleranceConfig[] = [];
+    for (var cad of config.split(';')) {
+      tolerances.push(parseAnnotation(cad));
+    }
+
+    return tolerances;
+  }
+  return getRateHealthConfigBy(ns, name, kind).tolerance;
+};
+
+const isNumber = (value: string | number): boolean => {
+  return value != null && value !== '' && !isNaN(Number(value.toString()));
+};
+
+export const parseAnnotation = (tolerance: string): ToleranceConfig => {
+  const cads = tolerance.split(',');
+  var code = cads[0];
+  var degraded = DEFAULTCONF.degraded;
+  var failure = DEFAULTCONF.failure;
+  var protocol = DEFAULTCONF.protocol;
+  var direction = DEFAULTCONF.direction;
+  cads.splice(0, 1); // Remove code value
+
+  var len = cads.length;
+  // More information ?
+  if (len > 0) {
+    // Check other parameters
+    if (isNumber(cads[0])) {
+      // Defined degraded and maybe failure
+      degraded = Number(cads[0]);
+      if (len > 1 && isNumber(cads[1])) {
+        // Defined failure too
+        failure = Number(cads[1]);
+        cads.splice(0, 2);
+      } else {
+        cads.splice(0, 1);
+      }
+      len = cads.length;
+    }
+    if (len > 0) {
+      // Defined Protocol or direction
+      if (len > 1) {
+        direction = new RegExp(cads[1]);
+      }
+      protocol = new RegExp(cads[0]);
+    }
+  }
+  const regex = code.toLowerCase().replace('x', 'd');
```

Comment:
Commented in the test, I think this miss the `\\d` format.
Locally fetched it seems to not work as expected.